openshift/deploy.sh

```diff
@@ -1,163 +1,35 @@
-    oc apply -f secrets.yaml
-}
-
-function oc_process_apply() {
-    echo -e "\\n Processing template - $1 ($2) \\n"
-    oc process -f $1 $2 | oc apply -f -
-}
-
-function openshift_login() {
-    oc login "${OC_URI}" -u "${OC_USERNAME}" -p "${OC_PASSWD}" --insecure-skip-tls-verify=true
-    if oc get project "${OC_PROJECT}"; then
-        oc project "${OC_PROJECT}"
-        echo "Removing all openshift resources from selected project"
-        oc delete all,cm,secrets --all
-    else
-        oc new-project "${OC_PROJECT}"
-    fi
-}
-
-function tag_rds_instance() {
-    TAGS="Key=ENV,Value=${DEPLOYMENT_PREFIX}"
-
-    echo "Tagging RDS instance with ${TAGS}"
-
-    aws rds add-tags-to-resource \
-            --resource-name "${RDS_ARN}" \
-            --tags "${TAGS}"
-}
-
-function get_rds_instance_info() {
-    aws --output=table rds describe-db-instances --db-instance-identifier "${RDS_INSTANCE_NAME}" 2>/dev/null
-}
-
-function allocate_aws_rds() {
-    if ! get_rds_instance_info; then
-        aws rds create-db-instance \
-        --allocated-storage "${RDS_STORAGE}" \
-        --db-instance-identifier "${RDS_INSTANCE_NAME}" \
-        --db-instance-class "${RDS_INSTANCE_CLASS}" \
-        --db-name "${RDS_DBNAME}" \
-        --db-subnet-group-name "${RDS_SUBNET_GROUP_NAME}" \
-        --engine postgres \
-        --engine-version "9.6.1" \
-        --master-username "${RDS_DBADMIN}" \
-        --master-user-password "${RDS_PASSWORD}" \
-        --publicly-accessible \
-        --storage-type gp2
-        #--storage-encrypted
-        echo "Waiting (60s) for ${RDS_INSTANCE_NAME} to come online"
-        sleep 60
-        wait_for_rds_instance_info
-    else
-        echo "DB instance ${RDS_INSTANCE_NAME} already exists, recreating database"
-        wait_for_rds_instance_info
-        PGPASSWORD="${RDS_PASSWORD}" psql -d template1 -h "${RDS_ENDPOINT}" -U "${RDS_DBADMIN}" -c "drop database ${RDS_DBNAME}"
-        PGPASSWORD="${RDS_PASSWORD}" psql -d template1 -h "${RDS_ENDPOINT}" -U "${RDS_DBADMIN}" -c "create database ${RDS_DBNAME}"
-    fi
-    tag_rds_instance
-}
-
-function wait_for_rds_instance_info() {
-    while true; do
-        echo "Trying to get RDS DB endpoint for ${RDS_INSTANCE_NAME} ..."
-
-        RDS_ENDPOINT=$(get_rds_instance_info | grep -w Address | awk '{print $4}')
-        RDS_ARN=$(get_rds_instance_info | grep -w DBInstanceArn | awk '{print $4}')
-
-        if [ -z "${RDS_ENDPOINT}" ]; then
-            echo "DB is still initializing, waiting 30 seconds and retrying ..."
-            sleep 30
-        else
-            break
-        fi
-    done
-}
+#Check if required env variables are set
+is_set_or_fail RDS_PASSWORD ${RDS_PASSWORD}
+is_set_or_fail RDS_INSTANCE_NAME ${RDS_INSTANCE_NAME}
+is_set_or_fail OC_USERNAME ${OC_USERNAME}
+is_set_or_fail OC_PASSWD ${OC_PASSWD}
+is_set_or_fail AWS_ACCESS_KEY_ID ${AWS_ACCESS_KEY_ID}
+is_set_or_fail AWS_SECRET_ACCESS_KEY ${AWS_SECRET_ACCESS_KEY}
```

Comment:
[Shellcheck](https://www.shellcheck.net/) says:
```
Line 27:
is_set_or_fail RDS_PASSWORD ${RDS_PASSWORD}
                            ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 28:
is_set_or_fail RDS_INSTANCE_NAME ${RDS_INSTANCE_NAME}
                                 ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 29:
is_set_or_fail OC_USERNAME ${OC_USERNAME}
                           ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 30:
is_set_or_fail OC_PASSWD ${OC_PASSWD}
                         ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 31:
is_set_or_fail AWS_ACCESS_KEY_ID ${AWS_ACCESS_KEY_ID}
                                 ^-- SC2086: Double quote to prevent globbing and word splitting.
 
Line 32:
is_set_or_fail AWS_SECRET_ACCESS_KEY ${AWS_SECRET_ACCESS_KEY}
                                     ^-- SC2086: Double quote to prevent globbing and word splitting.
```

Reply:
> Few schellcheck warnings

Nice, I didn't know about the website 👍 Thanks